onnxruntime/core/session/inference_session.h

```diff
@@ -172,6 +172,14 @@ class InferenceSession {
     */
   common::Status Initialize();
 
+  /**
+    * Initializes with a Model instance. Initialization includes but is not
+    * limited to graph transformations, construction of kernels, etc.
+	* @param model Model to use. InferenceSession::Load must not have been called previously.
+    * @return OK if success. Error information if failure.
+    */
+  common::Status Initialize(std::shared_ptr<Model>& model);
```

Comment:
Scott, FYI: We removed this method from the interface after RS5 due to protobuf version mismatch issues.

Reply:
So what alternate approach is best to avoid current ugly necessity to have to serialize a dynamically created Model and reload it in order to be able to execute it? Internal usage only. 

e.g. #168 constant_folding.cc dynamically creates a model but is forced to serialize it to use the current API.

Do we need to refactor to split out internal aspects from InferenceSession so that those can be called directly in this sort of scenario instead of using the public InferenceSession API?

Reply:
Removed from this PR. Constant folding is going to be refactored and shouldn't need an API change in InferenceSession.